python_modules/dagster/dagster/_core/definitions/asset_dep.py

```diff
@@ -54,6 +55,16 @@ def __new__(
     ):
         if isinstance(asset, AssetSpec):
             asset_key = asset.asset_key
+        elif isinstance(asset, AssetsDefinition):
+            # Only AssetsDefinition with a single asset can be passed
+            if len(asset.keys) > 1:
+                raise DagsterInvalidDefinitionError(
+                    "Cannot pass a multi_asset AssetsDefinition as an argument to deps."
+                    " Instead, specify dependencies on the assets created by the multi_asset"
+                    f" via AssetKeys or strings. For the multi_asset {asset.node_def.name}, the"
+                    f" available keys are: {asset.keys}."
+                )
+            asset_key = AssetKey.from_coercible_or_definition(asset)
         else:
```

Comment:
I think we should allow multiple key `AssetsDefinition` wherever we use `deps`. Could you address that in a follow up?

Reply:
yeah, will do